test/e2e/framework/test_context.go

```diff
@@ -32,6 +32,7 @@ func registerFlags(t *TestContextType) {
 	flag.StringVar(&t.KubeConfig, "kubeconfig", os.Getenv("KUBECONFIG"),
 		"Path to kubeconfig containing embedded authinfo.")
 	flag.Var(&TestContext.KubeContexts, "dp-context", "kubeconfig context for dataplane clusters (use several times).")
+	flag.Var(&TestContext.KubeContexts, "broker-context", "kubeconfig context for broker cluster.")
```

Comment:
Seems fine, 

My only worry is that we should not use the "broker" cluster for dataplane testing, it's fine for checking "Cluster" or "Endpoint" CRs, but not for submariner pods, dataplane, etc...

Keep that in mind, you can use ClusterC for that (probably passing -dp-context a third time)

Reply:
That would mean running e2e with total of 4 clusters so we thought of using broker cluster itself.

Reply:
What's the reason we shouldn't use the broker cluster for dataplane testing? We're not requiring the broker be it's own separate cluster so it will likely also have submariner installed in real environments. 

We could also add a 4th cluster but were concerned of the extra overhead in terms of resources.

Reply:
Another way to look at this is thinking we have 3 dataplane clusters  which broker installed on one of them :)

Reply:
I think we should use a separate var, BrokerContext

Reply:
Actually appending won't work properly unless the 3rd _-dp-context_ is specified. I think we should just use a fixed size `ClusterClients` array based on the `ClusterIndex` enum. In `Framework.BeforeEach`:

```
f.ClusterClients = make([]string, Broker + 1)
for i, context := range TestContext.KubeContexts {
    f.ClusterClients[i] = f.createKubernetesClient(context)
}	
ClusterClients[Broker] = f.createKubernetesClient(TestContext.BrokerContext)
```

In `TestContext.validateFlags`, check the allowed max length of `TestContext.KubeContexts`:

```
if len(t.KubeContexts) < (int(ClusterC) + 1) {
    klog.Fatalf("Too many dp-context parameters (%d) specified", len(t.KubeContexts))
}
```

Reply:
After thinking about this some more, I think we should just use pass _cluster1_ as the 3rd _dp-context_ so it's mapped to `ClusterC` for the add/remove  cluster tests.  The test can do a sanity check to ensure there's initially no gateway nodes configured. This way the code stays as is although I think we should still validate the max length of `KubeContexts` as outlined above.